File: 2404-most-frequent-even-element/2404-most-frequent-even-element.java
class Solution {
    public int mostFrequentEven(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        int ans = -1, freq = 0;
        for(int i: nums) {
            if(i%2 == 0) {
                map.put(i, map.getOrDefault(i, 0)+1);
                int f = map.getOrDefault(i, 0);
                if(f>freq) {
                    freq = f;
                    ans = i;
                } else if(f==freq) {
                    ans = Math.min(ans, i);
                }
            }
        }
        return ans;
    }
}

File: 2404-most-frequent-even-element/2404-most-frequent-even-element.cpp
class Solution {
    public:
    int mostFrequentEven(vector<int> &nums) {
        int mx=-1,res=-1;
    map<int,int>um;
        for(int i=0;i<nums.size();i++){
            um[nums[i]]++;
        }
        for(auto x:um){
            if(x.first%2==0 && x.second>mx){
                mx=x.second;
                res=x.first;
            }
        }
        return res;
    }
};

